public/js/scripts/contentCaching.js

```diff
@@ -0,0 +1,95 @@
+/*
+
+I am leaving this here so you could reuse some parts of it. Be aware of what you are doing
```

Comment:
That content caching is not entirely finished, still needs to be reviewed by Tomek and then I will remove it if we don't need it, or remove comments from parts that we need